src/utils/user.authentication.ts

```diff
@@ -11,10 +11,19 @@ import * as isemail from 'isemail';
 import {HttpErrors} from '@loopback/rest';
 import {UserProfile} from '@loopback/authentication';
 import {compare} from 'bcryptjs';
+import {hash} from 'bcryptjs';
 const compareAsync = promisify(compare);
 const jwt = require('jsonwebtoken');
 const signAsync = promisify(jwt.sign);
 const verifyAsync = promisify(jwt.verify);
+const hashAsync = promisify(hash);
+
+export async function hashPassword(
+  password: string,
+  salt: number,
+): Promise<string> {
+  return await hashAsync(password, salt);
+}
```

Comment:
Out of scope for this PR: to be ideal, such utility functions should be bound to the context to support customization.

Reply:
@raymondfeng Sounds reasonable to me, I have a PR for refactoring utilities into service:
https://github.com/strongloop/loopback4-example-shopping/pull/33
> utility functions should be bound to the context to support customization.

I think that PR is a good place to do ^